easybuild/toolchains/compiler/gcc.py

```diff
@@ -63,11 +64,21 @@ class Gcc(Compiler):
         'veryloose': ['mrecip=all', 'mno-ieee-fp'],
     }
 
-    COMPILER_OPTIMAL_ARCHITECTURE_OPTION = {
-        systemtools.AMD : 'march=native',
-        systemtools.INTEL : 'march=native',
-        systemtools.POWER: 'mcpu=native',  # no support for march=native on POWER
-    }
+    optarch = build_option('optarch')
+    if optarch == 'GENERIC':
```

Comment:
this should be a constant named `GENERIC`, that you import from `easybuild.tools.toolchain.toolchain`